libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1395,8 +1411,14 @@ void WaveTrack::PasteWaveTrack(double t0, const WaveTrack* other)
                     };
                 }
             }
-            if (auto *pClip = other->GetClipByIndex(0))
-               insideClip->Paste(t0, *pClip);
+            if (auto *pClip = other->GetClipByIndex(0)) {
+               bool success = insideClip->Paste(t0, *pClip);
+               // TODO wide wave tracks -- prove success, or propagate failure,
```

Comment:
> I can't strengthen the precondition -- because the function is used by WaveClip::Paste() which overrides Track::Paste(). Overriding functions can't strengthen preconditions (that violates the L principle of SOLID), and to state the precondition in Track would require the notion of width to be general for Tracks. I don't want that.

Right, I wasn't realizing that this is actually an overridden function